### PR TITLE
Cmr 7084 cloud stac dynamic catalog

### PR DESCRIPTION
**Pull request by sxu123**

### CODE REVIEW

File: search/lib/api/wfs.js [hunk 1]
```diff
@@ -111,8 +146,37 @@ async function getCollection (request, response) {
     await assertValid(schemas.collection, collectionResponse);
     response.json(collectionResponse);
   } catch (err) {
-    response.status(404).json(`Collection ${collectionId} not found for provider ${providerId}`);
+    response.status(404).json(`Error [${err}] occured when getting Collection ${collectionId} for provider ${providerId}`);
```
Comment (jaybarra) [inline search/lib/api/wfs.js:149]: typo: occured -> occurred

File: search/lib/api/wfs.js [hunk 2]
```diff
@@ -143,8 +207,38 @@ async function getGranules (request, response) {
...
+      const postSearchParams = new URLSearchParams(cmrParams);
+      if (allCloudCollections.length !== 0) {
+        for (i = 0; i < allCloudCollections.length; i++) {
+          postSearchParams.append("collection_concept_id", allCloudCollections[i]);
+        }
+      } else {
+        return response.status(400).json(`Cloud holding collections not found for provider [${providerId}].`);
+      }
+      if (concept_ids) {
+        for (i = 0; i < concept_ids.length; i++) {
```
Comment (jaybarra) [inline search/lib/api/wfs.js:234]: style change, prefer `concept_ids.forEach(` over `for`

**Review by jaybarra — APPROVED** [review]
generally prefer `.forEach` over `for` loops for handling arrays when every element is accessed

File: search/lib/api/wfs.js (hunk 1 above)
Comment (sxu123) [reply search/lib/api/wfs.js:149]: Done.

File: search/lib/api/wfs.js (hunk 2 above)
Comment (sxu123) [reply search/lib/api/wfs.js:234]: Done.

**Review by matthewhanson — APPROVED** [review]

**Comment by matthewhanson** [comment]
@sxu123 Looks good, I pushed a commit with some linter udates/fixes (semi-colons, quote usage, camelCase variables)

**Pull request merged**